src/Relation/Nullary.agda

```diff
@@ -23,11 +24,29 @@ infix 3 ¬_
 ¬_ : ∀ {ℓ} → Set ℓ → Set ℓ
 ¬ P = P → ⊥
 
+-- `Reflects` idiom.
+-- The truth value of P is reflected by a boolean value.
+
+data Reflects {p} (P : Set p) : Bool → Set p where
+  true  : ( p :   P) → Reflects P true
+  false : (¬p : ¬ P) → Reflects P false
+
 -- Decidable relations.
+-- This version of `Dec` allows the boolean portion of the
+-- value to compute independently from the proof portion.
+-- This often allows good computational properties when we
+-- only care about the boolean portion.
+
+record Dec {p} (P : Set p) : Set p where
+  constructor dec
+  field
```

Comment:
For the constructor name, may I suggest `mkDec`? Or maybe `_because_`?

Reply:
Yeah, I guess `mkDec` would work fine. I'm worried about the name getting long, because `dec false _` appears quite a lot and is already very long. I don't think `_because_` fits exactly; `true because (true ...)` doesn't read well.